autogluon/utils/tabular/features/auto_ml_feature_generator.py

```diff
@@ -84,22 +77,29 @@ def generate_features(self, X: DataFrame):
             X_categoricals = X[self.feature_transformations['category']]
             # TODO: Add stateful categorical generator, merge rare cases to an unknown value
             # TODO: What happens when training set has no unknown/rare values but test set does? What models can handle this?
-            if 'text' in self.feature_type_family:
-                self.feature_type_family_generated['text_as_category'] += self.feature_type_family['text']
+            if not self.fit:
+                if 'text' in self.feature_type_family:
+                    self.feature_type_family_generated['text_as_category'] += self.feature_type_family['text']
             X_categoricals = X_categoricals.astype('category')
             X_features = X_features.join(X_categoricals)
 
         if self.feature_transformations['text_special']:
+            X_text_special_combined = []
+            for nlp_feature in self.feature_transformations['text_special']:
+                X_text_special = self.generate_text_special(X[nlp_feature], nlp_feature)
+                X_text_special_combined.append(X_text_special)
+            X_text_special_combined = pd.concat(X_text_special_combined, axis=1)
             if not self.fit:
-                self.features_binned += list(X_text_features_combined.columns)
-                self.feature_type_family_generated['text_special'] += list(X_text_features_combined.columns)
-            X_features = X_features.join(X_text_features_combined)
+                self.features_binned += list(X_text_special_combined.columns)
+                self.feature_type_family_generated['text_special'] += list(X_text_special_combined.columns)
+            X_features = X_features.join(X_text_special_combined)
 
         if self.feature_transformations['datetime']:
             for datetime_feature in self.feature_transformations['datetime']:
                 X_features[datetime_feature] = pd.to_datetime(X[datetime_feature])
                 X_features[datetime_feature] = pd.to_numeric(X_features[datetime_feature])  # TODO: Use actual date info
-                self.feature_type_family_generated['datetime'].append(datetime_feature)
+                if not self.fit:
+                    self.feature_type_family_generated['datetime'].append(datetime_feature)
```

Comment:
Can you confirm all models, like the NeuralNet, are still actually utilizing the datetime feature after these changes?

Reply:
This does not affect downstream models, it only means that the feature name is appended to self.feature_type_family_generated['datetime'] during training, instead of at every call.

So if you have a datetime feature 'date', then after fit

`self.feature_type_family_generated['datetime'] == ['date']`

if you called predict 5 times:

`self.feature_type_family_generated['datetime'] == ['date', 'date', 'date', 'date', 'date', 'date']`

This wouldn't impact anything except take increasingly more memory to store the long list of repeated 'date' strings, particularly important if doing live-inference over an extended period of time on single-rows. This is the defect I fixed (same with text_as_category).